Unable to build master with external XROOTD

### Describe the bug
When trying to build with an external XRootD, cmake says it finds it but then it says it doesn't and this is making a spack build fail. Relevant lines in the cmake output:
```
-- Looking for XROOTD
-- Found Xrootd version num: 5.5.4.post256 (setting -DROOTXRDVERS=5.5.4.po)
-- Setting OLDPACK TRUE
-- XROOTD not found, enabling 'builtin_xrootd' option
-- Downloading and building XROOTD version 5.5.2
```
When building with XROOTD from spack (where I found this issue) the error is more verbose but I think is the same issue:
```
     121    -- Looking for XROOTD
     122    -- Found Xrootd version num: 5.5.3 (setting -DROOTXRDV
            ERS=500050003)
     123    --              libXrdMain not found: xproofd will be 
            a wrapper around xrootd
     124    --              libXrdClient not found: use built-in
  >> 125    CMake Error at cmake/modules/SearchInstalledSoftware.c
            make:992 (message):
     126      XROOTD not found.  Set environment variable XRDSYS t
            o point to your XROOTD
     127      installation, or include the installation of XROOTD 
            in the
     128      CMAKE_PREFIX_PATH.  Alternatively, you can also enab
            le the option
     129      'builtin_xrootd' to build XROOTD internally
```


### To Reproduce
```
# To install xrootd
git clone https://github.com/xrootd/xrootd
cd xrootd
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=../install
```

```
# To install ROOT
git clone https://github.com/root-project/root
cd root
mkdir builddir
cd builddir
cmake .. -DCMAKE_PREFIX_PATH=/path/to/xrootd/install  -Dbuiltin_xrootd=OFF -Dxrootd=ON -DCMAKE_VERBOSE_MAKEFILE=ON
```

### Setup
cmake 3.26.2
OS: Arch Linux

### Additional context
I'm posting it here since I tried different versions of XRootD and all failed, maybe it's an issue with XROOTD since it's complaining about not finding libXrdMain nor libXrdClient

--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -988,7 +988,7 @@ endforeach()
 
 if(xrootd AND NOT builtin_xrootd)
   message(STATUS "Looking for XROOTD")
-  find_package(XROOTD)
+  find_package(XRootD) # Case sensitivity is important here to match the XRootDConfig.cmake file
   if(NOT XROOTD_FOUND)
     if(fail-on-missing)
       message(FATAL_ERROR "XROOTD not found. Set environment variable XRDSYS to point to your XROOTD installation, "

